tool/gh_actions/update_directory.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# set_up.sh
+# GitHub Actions step: Set up the files directory
+#
+# 2023 May 3
+# Author: Yao Jing Quek <yao.jing.quek@intel.com>
+
+set -euo pipefail
+
+root_directory=$(git rev-parse --show-toplevel)
+
+src_path="$root_directory/src"
+rohd_submodule_path="$root_directory/rohd"
+
+git submodule update --init --recursive
+git submodule update --remote rohd
```

Comment:
If a document name or directory structure in the `rohd` repo changes in a way that breaks `rohd-website`'s automation, then this could break the website, right?  Should we update the submodule via a cron-created pull request instead of immediately re-updating the website?

This article has some cool inspiration for how GitHub itself updates it's dependencies automatically by generating pull requests: https://github.blog/2023-04-06-building-github-with-ruby-and-rails/

We could probably make it so that it only opens the pull request if the documentation relative to the website changes (maybe as simple as `git diff`)?

Reply:
Correct me if I am wrong, the flow going to be:

1. git pull and update the submodule
2. **IF** git diff is not quiet (got changes)
  2.1 Delete all the directory in destination
  2.2 Recopy all the files from src into the destination
  2.3 GitHub trigger pull request
  2.4 Admin receive notification and approve the pull request
  2.5 Once it merged into the main repository
       2.5.1 Retrigger the website deployment
4. **ELSE** (no changes)
   4.1. Nothing to be done

Reply:
yes, i think that sounds right!

Reply:
Pull request will be fail if new files being added, user need to manually update the `navigation.yml`